silq/pulses/pulse_types.py

```diff
@@ -668,6 +668,138 @@ def get_voltage(self, t: Union[float, Sequence]) -> Union[float, np.ndarray]:
+             'offset', 'frequency_sideband', 'sideband_mode', 'phase_reference'])
+
+        if self.sideband_mode is None:
+            self.sideband_mode = 'IQ'
+        if self.phase_reference is None:
+            self.phase_reference = 'absolute'
+        if self.offset is None:
+            self.offset = 0
+
+    def __repr__(self):
+        properties_str = ''
+        try:
+            properties_str = f'f_LO={freq_to_str(self.frequency)}'
+            properties_str += f', f={self.frequencies} Hz'
+            properties_str += f', phases={self.phases} deg'
+            properties_str += '(rel)' if self.phase_reference == 'relative' else '(abs)'
+
+            if self.power is not None:
+                properties_str += f', power={self.power} dBm'
+            if self.amplitude is not None:
+                properties_str += f', A_LO={self.amplitude} V'
+            if self.amplitudes is not None:
+                properties_str += f', A={self.amplitudes} V'
+            if self.offset:
+                properties_str += f', offset={self.offset} V'
+            if self.frequency_sideband is not None:
+                properties_str += f'f_sb={freq_to_str(self.frequency_sideband)} ' \
+                                  f'{self.sideband_mode}'
+
+            properties_str += f', t_start={self.t_start}'
+            properties_str += f', duration={self.duration}'
+        except:
+            pass
+
+        return super()._get_repr(properties_str)
+
+    def get_voltage(self, t: Union[float, Sequence]) -> Union[float, np.ndarray]:
+        """Get voltage(s) at time(s) t.
+
+        Raises:
+            AssertionError: not all ``t`` between `Pulse`.t_start and
+                `Pulse`.t_stop
+        """
+        assert is_between(t, self.t_start, self.t_stop),\
+            f"voltage at {t} s is not in the time range " \
+            f"{self.t_start} s - {self.t_stop} s of pulse {self}"
+
+        if self.phase_reference == 'relative':
+            t = t - self.t_start
+
+        assert self.amplitudes is not None, f'Pulse {self.name} does not have specified amplitudes.'
+        assert self.frequencies is not None, f'Pulse {self.name} does not have specified frequencies.'
+        assert self.phases is not None, f'Pulse {self.name} does not have specified phases.'
+
+        assert len(self.amplitudes) == len(self.frequencies) == len(self.phases), f'Pulse {self.name} does not have '\
+                                                                                  f'equal number of amplitudes, ' \
+                                                                                  f'frequencies and phases.'
+        waveform = 0.0
+        for amp, freq, phase in zip(self.amplitudes, self.frequencies, self.phases):
+            waveform += amp * np.sin(2 * np.pi * (freq * t + phase / 360))
+        waveform += self.offset
+        # We need to insure the waveform is limited by +- 1V:
+        waveform = waveform/len(self.frequencies)
```

Comment:
I don't agree with this line. The pulse itself shouldn't by default be limited to +-1V, this is only the case if it is used for IQ modulation. If I create the sum of two Sine pulses, each with an amplitude of 1V, I don't expect each pulse to suddenly have an amplitude of 0.5V

Reply:
True, I will move these conditions into interface

Reply:
corrected.